### PR TITLE
Fix bug 1151436. Handle signatures over 255 chars correctly.

### PR DESCRIPTION
**Pull request by jkozera**
Runs SigTrunc before SignatureRunWatchDog to truncate signatures with "shutdownhang" correctly. Also adds a simple test for mozilla_processor_2015.py.

### CODE REVIEW

**Comment by rhelmer** [comment]
Thanks, this lgtm!

However does it really fix bug 1151436 ? It seems like it puts in a test that shows that processor is not causing the problem (which is great!)

I think the problem in bug 1151436 is that our ```reports``` table in postgres has a limit on the ```signature``` of 256 characters (```varchar(256)```)

One fix for that would be to make it ```text``` instead, but I don't know the implications of altering the reports table (it's large and has many partitions and is pretty central to our current reporting system so a bit hard to test.)

I am going to ask @AdrianGaudebert in the bug if we're planning to re-land the Elasticsearch backend for the ```/report/list``` page, which should fix this as well.

File: socorro/unittest/processor/test_mozilla_processor_2015.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+            raw_crash,
+            raw_dumps,
+            DotDict()
+        )
+
+        ok_(processed_crash.success)
+        eq_(processed_crash.processor_notes,
+            'dwight; MozillaProcessorAlgorithm2015; '
+            'SignatureTool: signature truncated due to length; '
+            'SignatureTool: signature truncated due to length')
```
Comment (twobraids) [inline socorro/unittest/processor/test_mozilla_processor_2015.py]: this is interesting - I'm not sure, off hand, why the processor note about truncation is repeated.   Looking in production database, I can only see that condition when a crash is reprocessed, and in that situation, the two messages are not contiguous.    I'm going to spend some time investigating why it repeats here.

Comment (twobraids) [inline socorro/unittest/processor/test_mozilla_processor_2015.py]: ah, the signature generator, a class that predates the rule system has a truncation routine in it. So a long signature is truncated on generation.  Then other rules lengthen the signature, and then the SigTrunc rule truncates it a second time.  Not ideal, but out of the scope of this bug to repair.

**Comment by twobraids** [comment]
r+  
thank you very much for this patch.  You did a great job.  I'll be merging this into the code base later today.

**Comment by twobraids** [comment]
I've created http://bugzil.la/1155165 for the duplication issue.  @jkozera if you have an interest, you are invited to look at that bug, too.  Don't feel obligated, it'll get resolved by someone eventually.  

Again, thank you for the resolution of this problem.

**Pull request merged**